### PR TITLE
fix(backend): require all checks to pass

### PR DESCRIPTION
**Pull request by haraldh**
And error out with a message to run `enarx info` to prevent users from
getting strange error messages later.

Signed-off-by: Harald Hoyer <harald@profian.com>

### CODE REVIEW

File: src/backend/kvm/mod.rs [hunk 1]
```diff
@@ -80,7 +80,7 @@ impl crate::backend::Backend for Backend {
 
     #[inline]
     fn have(&self) -> bool {
-        data::dev_kvm().pass
+        self.data().iter().fold(true, |acc, x| acc & x.pass)
```
Comment (bstrie) [inline src/backend/kvm/mod.rs:83]: How about `self.data().iter().all(|x| x.pass)`.

File: src/backend/sev/mod.rs [hunk 2]
```diff
@@ -55,7 +55,7 @@ impl super::Backend for Backend {
 
     #[inline]
     fn have(&self) -> bool {
-        data::dev_sev_writable().pass
+        self.data().iter().fold(true, |acc, x| acc & x.pass)
```
Comment (bstrie) [inline src/backend/sev/mod.rs:58]: Same here.

File: src/backend/sgx/mod.rs [hunk 3]
```diff
@@ -39,7 +39,7 @@ impl crate::backend::Backend for Backend {
 
     #[inline]
     fn have(&self) -> bool {
-        data::dev_sgx_enclave().pass
+        self.data().iter().fold(true, |acc, x| acc & x.pass)
```
Comment (bstrie) [inline src/backend/sgx/mod.rs:42]: Same here.

File: src/backend/kvm/mod.rs (hunk 1 above)
Comment (haraldh) [reply src/backend/kvm/mod.rs:83]: even better .. thanks

Comment (haraldh) [reply src/backend/kvm/mod.rs]: done

File: src/backend/sgx/mod.rs (hunk 3 above)
Comment (haraldh) [reply src/backend/sgx/mod.rs]: done

File: src/backend/sev/mod.rs (hunk 2 above)
Comment (haraldh) [reply src/backend/sev/mod.rs]: done

**Review by rvolosatovs — APPROVED** [review]

**Pull request merged**